src/theory/difficulty_manager.cpp

```diff
@@ -51,11 +51,15 @@ void DifficultyManager::getDifficultyMap(std::map<Node, Node>& dmap)
   }
 }
 
-void DifficultyManager::notifyLemma(Node n)
+void DifficultyManager::notifyLemma(Node n, bool inFullEffortCheck)
 {
-  if (options::difficultyMode() != options::DifficultyMode::LEMMA_LITERAL)
+  if (options::difficultyMode() != options::DifficultyMode::LEMMA_LITERAL_ALL)
   {
-    return;
+    if (!inFullEffortCheck
+        || options::difficultyMode() != options::DifficultyMode::LEMMA_LITERAL)
+    {
+      return;
+    }
```

Comment:
I think it be simpler to test that you return when `!inFullEffortCheck && options::difficultyMode() == options::DifficultyMode::LEMMA_LITERAL` holds.